packages/trader/src/AppV2/Components/TradeParameters/RiskManagement/take-profit-and-stop-loss-container.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { observer } from 'mobx-react';
+import { useTraderStore } from 'Stores/useTraderStores';
+import { Button, useSnackbar } from '@deriv-com/quill-ui';
+import { localize, Localize } from '@deriv/translations';
+import { WS } from '@deriv/shared';
+import { getSnackBarText } from 'AppV2/Utils/trade-params-utils';
+import TakeProfitAndStopLossInput from './take-profit-and-stop-loss-input';
+
+type TTakeProfitAndStopLossContainerProps = {
+    closeActionSheet: () => void;
+    should_show_deal_cancellation?: boolean;
+};
+
+const TakeProfitAndStopLossContainer = observer(({ closeActionSheet }: TTakeProfitAndStopLossContainerProps) => {
+    const {
+        has_take_profit,
+        has_cancellation,
+        has_stop_loss,
+        take_profit,
+        onChangeMultiple,
+        stop_loss,
+        validation_errors,
+    } = useTraderStore();
+
+    const { addSnackbar } = useSnackbar();
+
+    const [tp_error_text, setTPErrorText] = React.useState(validation_errors?.take_profit?.[0] ?? '');
```

Comment:
why are we storing subscription_ids? 

Also, onSave has insane amount of logic. Dont know whats going on.

Maybe we should ask  to make the component behavior simpler instead of hacking up a solution to make it happen

Reply:
We are storing subscription_ids in ref in order to have an opportunity to clean it

Reply:
Regarding amount of logic: yes, I agree. I wrote to Slack regarding simplifying. Also, we are waiting for some improvements from BE side, like parallel validation of TP&SL and  sending  validation params in error too.
Currently, logic is so complicated, because I need to send parallel proposal for getting validat. params and get errors if any. Also it case if value is closing ActionSheet without saving, I need to restore previous values

Reply:
We should negotiate with design in this case and also should have been blocked by backend accordingly or even partially develop it without validation.

Reply:
We already asked BE team for changes, but it will take some time. In order not to block our FE development, I decided to proceed with this PR after consultation with Bala and Maryia